R/boundsPostprob.R

```diff
@@ -1,72 +1,72 @@
-#' @return A matrix where for each sample size in \code{nvec}, this function
-#' returns the maximum number of responses that meet the futility
-#' threshold (xL), its corresponding response rate (pL), posterior probability
-#' (postL), upper bound of one sided 95% CI for the response rate based on an
-#' exact binomial test (UciL), and the same boundary parameters for efficacy:
-#' the minimal number of responses that meet the efficacy threshold (xU),
-#' the corresponding response rate (pU), posterior probability (postU) and
-#' the lower bound of one sided 95% CI for the response rate based on exact
-#' binomial test (LciU).
-#'
-#' @importFrom stats binom.test
+#' @inheritParams postprob
+#' @inheritParams ocPostprob
+#' @typed looks : numeric
+#'  A vector of number of patients in each look.
+#' @return A matrix for each same size in `looks`. For each sample size, the following is returned:
+#' - `xL` : the maximum number of responses that meet the futility.
+#'          threshold
+#' - `pL` : response rate corresponding to `xL`.
+#' - `postL`: posterior probability corresponding to `xL`.
+#' - `pL_upper_ci` : upper bound of one sided 95% CI for the response rate `pL` based on an
+#'            exact binomial test.
+#' - `xU` : the minimal number of responses that meet the efficacy threshold.
+#' - `pU` : response rate corresponding to `xU`.
+#' - `postU` : posterior probability corresponding to `xU`.
+#' - `pU_lower_ci` : lower bound of one sided 95% CI for the response rate `pU` based on exact
+#'            binomial test.
 #'
 #' @example examples/boundsPostprob.R
 #' @export
-#' @keywords graphics
-boundsPostprob <- function(nvec, p0, p1 = p0, tL, tU, a, b) {
-  z <- matrix(NA, length(nvec), 6)
-  dimnames(z) <- list(nvec, c(
-    "xL", "pL", "postL",
-    "xU", "pU", "postU"
-  ))
+boundsPostprob <- function(looks, p0, p1 = p0, tL, tU, parE, weights) {
+  z <- matrix(NA, nrow = length(looks), ncol = 8)
   znames <- c(
-    "xL", "pL", "postL", "UciL",
-    "xU", "pU", "postU", "LciU"
+    "xL", "pL", "postL", "pL_upper_ci",
+    "xU", "pU", "postU", "pU_lower_ci"
   )
-  z <- matrix(NA, length(nvec), length(znames))
-  dimnames(z) <- list(nvec, znames)
+  dimnames(z) <- list(looks, znames)
   k <- 0
-  for (n in nvec) {
+  parE <- t(parE)
+  if (missing(weights)) {
+    weights <- rep(1, nrow(parE))
+  }
+  for (n in looks) {
     k <- k + 1
     # initialize so will return NA if 0 or n in "continue" region
     xL <- NA
     xU <- NA
     for (x in 0:n) {
-      postp <- postprob(x, n, p1, parE = c(a, b))
-      if (postp <= tL) {
+      postp <- 1 - postprob(x, n, p0, parE, weights) # futility look
+      if (postp >= tL) { # Rule is P(RR < p0) > tL
+        postL <- postp
         xL <- x
       }
-      if (p0 != p1) {
-        postp <- postprob(x, n, p0, parE = c(a, b))
-      }
-      if (postp >= tU) {
+      postp <- postprob(x, n, p1, parE, weights) # efficacy look
```

Comment:
```suggestion
      postp_eff <- postprob(x, n, p1, parE, weights) # efficacy look
```
and adapt other code accordingly